src/main/webapp/group.js

```diff
@@ -0,0 +1,235 @@
+function createPostResult(result) {
+    var container = document.createElement("div");
+    var name = document.createElement("h5");
+    name.style.textTransform = 'capitalize';
+    var content = document.createElement("h4");
+    var replyForm = document.createElement("form");
+    var replyInput = document.createElement("input");
+    var replySubmit = document.createElement("button");
+    var viewThread = document.createElement("button");
+
+    if (result.userID != "anonymous") {
+        getUser(result.userID).then((user) => {
+            var userInfo;
+            // If the user is both a student and a tutor access the appropriate location to find informaton
+            if (user.student != null) {
+                userInfo = user.student;
+            } else {
+                userInfo = user;
+            }
+            name.innerText = "By " + userInfo.name;
+        });
+    } else {
+        name.innerText = "By " + result.userID;
+    }
+
+    replyForm.className = "form-inline my-2 my-lg-0";
+    replyInput.className = "form-control mr-sm-2";
+    replyInput.id = "reply-input";
+    replyInput.type = "search";
+    replyInput.placeholder = "Write your reply here";
+    replyInput.required = true;
+    replySubmit.className = "btn btn-outline-success my-2 my-sm-0";
+    replySubmit.type = "submit";
+    replySubmit.innerText = "Submit";
+    replyForm.addEventListener('submit', event => {
+        event.preventDefault();
+        addReply(result, document.getElementById('reply-input').value);  
+    });
+
+    content.innerText = result.content;
+
+    replyForm.appendChild(replyInput);
+    replyForm.appendChild(replySubmit);
+
+    viewThread.className = "btn btn-outline-success my-2 my-sm-0";
+    viewThread.innerText = "View Thread";
+    viewThread.addEventListener('click', () => {
+        displayThread(result, container);
+    });
+   
+
+    container.classList.add("user-result");
+    container.classList.add("list-group-item");
+
+    container.appendChild(content);
+    container.appendChild(name);
+    container.appendChild(replyForm);
+    container.appendChild(viewThread);
+
+    return container;
+}
+
+function addReply(result, content, window) {
+    const params = new URLSearchParams();
+
+    params.append('postId', result.id);
+    params.append('content', content);
+
+    fetch('/manage-replies', {method: 'POST', body: params}).then((response) => {
+        //if there was an error posting to the group
+        if(response.error) {
+            window.location.href = "/group.html?groupId=" + result.groupID;
+            alert("There was an error when posting to this group.");
+            return;
+        }
+        window.location.href = "/group.html?groupId=" + result.groupID;
+    });
+}
+
+async function displayThread(post, container, document) {
+    var replies = displayThreadHelper(post, container, document);
+    
+    await replies;
+}
```

Comment:
```suggestion
function displayThread(post, container, document) {
   return displayThreadHelper(post, container, document);
}
```

Reply:
I thought I had to use await here but I tested this suggestion and it works. Thanks!